src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -405,6 +405,10 @@ The URL is an identifier only. OpenSearch Dashboards and your browser will never
             )
           );
         }
+        // Converting the visibleVisLayers array back to a map
+        if (result.visibleVisLayers !== undefined && Array.isArray(result.visibleVisLayers)) {
+          result.visibleVisLayers = new Map<VisLayerTypes, boolean>(result.visibleVisLayers);
```

Comment:
NIT: I think this added logic assumes that the array are already in the correct format (tuples of [key, value]) so it initializes the `Map` with the `visibleVisLayers` array directly. Is it always true?

Reply:
Yes it is always true, since `visibleVisLayers` is only set when converting the `Map` to an array to begin with.